toolchain/check/import_cpp.cpp

```diff
@@ -44,6 +44,16 @@ static auto GenerateCppIncludesHeaderCode(
   std::string code;
   llvm::raw_string_ostream code_stream(code);
   for (const Parse::Tree::PackagingNames& import : imports) {
+    // Add a line marker directive pointing at the location of the `import Cpp`
+    // declaration in the Carbon source file. This will cause Clang's
+    // diagnostics machinery to track and report the location in Carbon code
+    // where the import was written.
+    auto import_loc = context.tokens().TokenToDiagnosticLoc(
+        context.parse_tree().node_token(import.node_id));
+    code_stream << "# " << import_loc.loc.line_number << " \"";
+    code_stream.write_escaped(import_loc.loc.filename);
+    code_stream << "\"\n";
```

Comment:
There's a helper for write_escaped:

```suggestion
    code_stream << "# " << import_loc.loc.line_number << " \"" << FormatEscaped(import_loc.loc.filename) << "\"\n";
```

Reply:
Done.